src/main/java/in/projecteka/gateway/session/SessionController.java

```diff
@@ -31,6 +34,15 @@
         return identityService.getTokenFor(session.getClientId(), session.getClientSecret());
     }
 
+    @PostMapping(USER_SESSION)
+    public Mono<Session> sessionFor(@RequestBody SessionRequest session) {
+        logger.info("Session request received {}", keyValue("username", session.getUserName()));
+        return identityServiceClient.getUserToken(session.getClientId(),
+                session.getClientSecret(),
+                session.getUserName(),
+                session.getPassword());
+    }
```

Comment:
why are you expecting them clientId and clientSecret? can't we take those from the config itself? then you will be sure that they under gateway client right?

Reply:
Done